### PR TITLE
documentloaders: add support for PDF loader

### PR DESCRIPTION
**Pull request by navicstein**
This PR adds the support for PDF documentloader, added a tiny pdf test file & `Load` test cases 

 
### PR Checklist

- [x] Read the [Contributing documentation](https://github.com/tmc/langchaingo/blob/main/CONTRIBUTING.md).
- [x] Read the [Code of conduct documentation](https://github.com/tmc/langchaingo/blob/main/CODE_OF_CONDUCT.md).
- [x] Name your Pull Request title clearly, concisely, and prefixed with the name of the primarily affected package you changed according to [Good commit messages](https://go.dev/doc/contribute#commit_messages) (such as `memory: add interfaces for X, Y` or `util: add whizzbang helpers`).
- [x] Check that there isn't already a PR that solves the problem the same way to avoid creating a duplicate.
-  [ ] Provide a description in this PR that addresses **what** the PR is solving, or reference the issue that it solves (e.g. `Fixes #123`).
- [x] Describes the source of new concepts.
- [ ] References existing implementations as appropriate.
- [x] Contains test coverage for new functions.
- [x] Passes all [`golangci-lint`](https://golangci-lint.run/) checks.

### CODE REVIEW

File: documentloaders/pdf_test.go [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+	assert.NoError(t, err)
+
+	loader := NewPDF(file)
+
+	docs, err := loader.Load(context.Background())
+	require.NoError(t, err)
+
+	expectedText := "Te quo illum phaedrum salutatus, has in quis alii vide."
+	assert.Contains(t, docs[0].PageContent, expectedText)
+	assert.Equal(t, docs[0].Metadata["Pages"], "1")
```
Comment (tobiade) [inline documentloaders/pdf_test.go:24]: You have arguments the wrong way round here. It should be: `assert.Equal(t, "1", docs[0].Metadata["Pages"])`

Comment (navicstein) [reply documentloaders/pdf_test.go]: fixed, I was copying test strategy from `TestCSVLoader`

**Comment by mplachter** [comment]
Looks like doconv uses exec.command for the PDF Conversion.

It looks like it takes the reader and than also makes a duplicate file before converting it.

https://github.com/sajari/docconv/blob/master/pdf_text.go#L28-L33

**Comment by tmc** [comment]
duped by https://github.com/tmc/langchaingo/pull/188

**Pull request closed without merging**